07-routing-gossip.md

```diff
@@ -1136,6 +1157,200 @@ A->D's `update_add_htlc` message would be:
 And D->C's `update_add_htlc` would again be the same as B->C's direct payment
 above.
 
+## Liquidity Ads
+
+Liquidity ads allow nodes to announce their willingness to provide funding to
+other nodes for a fee. Sellers advertise various rates for the liquidity they
+are selling depending on the guarantees offered to buyers.
+
+1. subtype: `will_fund_lease_rates`
+2. data:
+    * [`will_fund_lease_rates_tlvs`:`lease_rates`]
+
+1. `tlv_stream`: `will_fund_lease_rates_tlvs`
+2. types:
+    1. type: 1 (`basic_funding_leases`)
+    2. data:
+        * [`...*basic_funding_lease`:`funding_lease_rates`]
+    1. type: 3 (`duration_based_funding_leases`)
+    2. data:
+        * [`...*duration_based_funding_lease`:`funding_lease_rates`]
+
+1. subtype: `request_funds`
+2. data:
+    * [`u64`:`requested_sats`]
+    * [`byte`:`lease_type`]
+    * [`funding_lease`:`funding_lease`]
+
+1. `lease_type`: 1 (`basic_funding_lease`)
+2. data:
+    * [`u32`:`min_lease_amount_sat`]
+    * [`u32`:`max_lease_amount_sat`]
+    * [`funding_lease_fee`:`funding_lease_fee`]
+
+1. `lease_type`: 3 (`duration_based_funding_lease`)
+2. data:
+    * [`u16`:`lease_duration`]
+    * [`u32`:`min_lease_amount_sat`]
+    * [`u32`:`max_lease_amount_sat`]
+    * [`funding_lease_fee`:`funding_lease_fee`]
+    * [`u16`:`max_channel_fee_basis`]
+    * [`u32`:`max_channel_fee_base_msat`]
+
+1. subtype: `funding_lease_fee`
+2. data:
+    * [`u16`:`funding_weight`]
+    * [`u16`:`lease_fee_basis`]
+    * [`u32`:`lease_fee_base_sat`]
+
+1. subtype: `will_fund`
+2. data:
+    * [`byte`:`lease_witness_type`]
+    * [`lease_witness`:`lease_witness`]
+    * [`signature`:`signature`]
+
+1. `lease_witness_type`: 1 (`basic_funding_lease_witness`)
+2. data:
+    * [`u16`:`funding_script_size`]
+    * [`funding_script_size`:`funding_script`]
+
+1. `lease_witness_type`: 3 (`duration_based_funding_lease_witness`)
+2. data:
+    * [`u32`:`lease_expiry`]
+    * [`u16`:`funding_script_size`]
+    * [`funding_script_size`:`funding_script`]
+    * [`u16`:`max_channel_fee_basis`]
+    * [`u32`:`max_channel_fee_base_msat`]
+
+Sellers may offer multiple `lease_type`s, described in the following sections.
+Buyers select a specific lease offered by the seller and use `request_funds`
+to purchase that lease. Sellers answer with `will_fund` containing a signature
+that commits to the lease parameters included in the `lease_witness`.
+
+### The `basic_funding_lease` type
+
+A `basic_funding_lease` does not provide any guarantee that the seller won't
```

Comment:
I'm not sure what you mean, can you rephrase?

Reply:
We did discuss ways of creating guarantees for the buyer in https://lists.linuxfoundation.org/pipermail/lightning-dev/2023-December/004227.html

We can create lease types today that modify scripts to ensure that the seller cannot get their funds back immediately. I don't think that's a good idea, because of the drawbacks discussed in this email thread, but other people might feel differently and may want to ship such lease types.

So I'm not sure this deserves to be in a separate section, especially if we drop the `duration_based_funding_lease` from this initial version (which I'll do in a second commit, so that people can still see what it looks like to define another lease type in the git history).